wd_helper.au3

```diff
@@ -2196,6 +2196,44 @@ Func _WD_CheckContext($sSession, $bReconnect = Default, $vTarget = Default)
 	Return SetError(__WD_Error($sFuncName, ($iResult) ? $_WD_ERROR_Success : $_WD_ERROR_Exception), 0, $iResult)
 EndFunc   ;==>_WD_CheckContext
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_FindElement_ByRegExp
+; Description ...: Find element by matching attributes values using regular expression
+; Syntax ........: _WD_FindElement_ByRegExp($sSession, $sMode, $sRegEx[, $sRegExFlags = ""[, $bAll = False]])
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $sMode               - Attribute of the element which should be matched, e.g. `id`, `style`, `class` etc.
+;                  $sRegEx              - RegEx in JavaScript format
+;                  $sRegExFlags         - [optional] RegEx Flags. Default is "".
+;                  $bAll                - [optional] Fetch a list of all elements that fits to the RegEx pattern ? Default is False - return only first element
+; Return values .: Success - Matched element
+;                  Failure - Response from _WD_ExecuteScript() and sets @error to value returned from _WD_ExecuteScript()
+; Author ........: TheDcoder
+; Modified ......: mLipok
+; Remarks .......: The RegEx matching is done by the browser's JavaScript engine so AutoIt's RegEx rules may not accurately work
+;                  in this function. You may refer to this cheatsheet for further information:
+;                  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Cheatsheet
+;                  The matching elements are returned in the same format as the results from _WD_FindElement
```

Comment:
This is not currently how the function is operating. The resulting array contains objects instead of strings. The coding to handle this is already in place in _WD_FindElement and would need to be replicated here or moved to an internal function.

Reply:
Please take a look on: [returning as array](https://github.com/Danp2/au3WebDriver/pull/301/commits/b3a4be35cf5d87467ed5949c2960ee06d2a595c4)

Reply:
my testing script:
```autoit
Func _EXAMPLE__WD_FindElement_ByRegExp($sSesion)
	_WD_Navigate($sSesion, 'https://github.com/Danp2/au3WebDriver/commits/master')
	_WD_LoadWait($sSesion)
;~ 	data-url="/Danp2/au3WebDriver/commits/8bdf053f3b97d0b585b9f5bf2d6607a67eb5a545/commits_list_item"
	Local $vResult = _WD_FindElement_ByRegExp($sSesion, 'data-url', '\/danp2.au3webdr.*\/commits_list_item', 'i', False)
	If @error Then ConsoleWrite("! ---> @error=" & @error & "  @extended=" & @extended & _
			" : _WD_FindElement_ByRegExp" & @CRLF)

	_ArrayDisplay($vResult, '$vResult')
EndFunc   ;==>_EXAMPLE__WD_FindElement_ByRegExp
```

result:
![image](https://user-images.githubusercontent.com/11089482/169608791-3896fd74-70e7-4e28-97f7-e90600f4f646.png)

Reply:
it requires: [returning as array](https://github.com/Danp2/au3WebDriver/pull/301/commits/b3a4be35cf5d87467ed5949c2960ee06d2a595c4)